src/output-json-mysql.c

```diff
@@ -0,0 +1,175 @@
+/* Copyright (C) 2022-2024 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \author linqiankai <linqiankai@geweian.com>
+ *
+ * Implement JSON/eve logging for app-layer MySQL.
+ */
+
+#include "suricata-common.h"
+#include "detect.h"
+#include "pkt-var.h"
+#include "conf.h"
+
+#include "threads.h"
+#include "threadvars.h"
+#include "tm-threads.h"
+
+#include "util-unittest.h"
+#include "util-buffer.h"
+#include "util-debug.h"
+#include "util-byte.h"
+
+#include "output.h"
+#include "output-json.h"
+
+#include "app-layer.h"
+#include "app-layer-parser.h"
+
+#include "output-json-mysql.h"
+#include "rust.h"
+
+#define MYSQL_LOG_NULL  BIT_U32(0)
+#define MYSQL_DEFAULTS      (MYSQL_LOG_NULL)
+
+typedef struct OutputMysqlCtx_ {
+    uint32_t flags;
+    OutputJsonCtx *eve_ctx;
+} OutputMysqlCtx;
+
+typedef struct LogMysqlLogThread_ {
+    OutputMysqlCtx *mysqllog_ctx;
+    OutputJsonThreadCtx *ctx;
+} LogMysqlLogThread;
+
+static int JsonMysqlLogger(ThreadVars *tv, void *thread_data, const Packet *p, Flow *f, void *state,
+        void *txptr, uint64_t tx_id)
+{
+    LogMysqlLogThread *thread = thread_data;
+    SCLogDebug("Logging mysql transaction %" PRIu64 ".", tx_id);
+
+    JsonBuilder *jb =
+            CreateEveHeader(p, LOG_DIR_FLOW, "mysql", NULL, thread->mysqllog_ctx->eve_ctx);
+    if (unlikely(jb == NULL)) {
+        return TM_ECODE_FAILED;
+    }
+
+    if (!rs_mysql_logger(txptr, thread->mysqllog_ctx->flags, jb)) {
```

Comment:
You should not need this file now
`rs_mysql_logger` (which should rather be named `SCMysqlLogger`) cf `EveJsonSimpleTxLogFunc` in output.c

Reply:
Fixed, please check again.